src/Database/QueryCompiler.php

```diff
@@ -125,7 +125,9 @@ public function compile(Query $query, ValueBinder $generator)
     protected function _sqlCompiler(&$sql, $query, $generator)
     {
         return function ($parts, $name) use (&$sql, $query, $generator) {
-            if (!count($parts)) {
+            if (!isset($parts) ||
```

Comment:
As a follow on, $parts can be an integer (I've dumped out the type of parts so this is not a hypothetical,  sometimes an integer is passed in).  So the following case shows why empty would be a change from current behavior:

```PHP
$parts = 0;
!count($parts)   // == 1 or effectively true
empty($parts)    // == false
```